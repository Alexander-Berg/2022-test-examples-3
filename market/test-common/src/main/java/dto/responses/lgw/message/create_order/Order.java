package dto.responses.lgw.message.create_order;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Order {

    @JsonProperty("cargoType")
    private Integer cargoType;

    @JsonProperty("orderId")
    private OrderId orderId;

    @JsonProperty("amountPrepaid")
    private Integer amountPrepaid;

    @JsonProperty("documentData")
    private DocumentData documentData;

    @JsonProperty("warehouseFrom")
    private WarehouseFrom warehouseFrom;

    @JsonProperty("total")
    private Integer total;

    @JsonProperty("tariff")
    private String tariff;

    @JsonProperty("shipmentDate")
    private String shipmentDate;

    @JsonProperty("delivery")
    private Delivery delivery;

    @JsonProperty("cargoCost")
    private Integer cargoCost;

    @JsonProperty("assessedCost")
    private Integer assessedCost;

    @JsonProperty("deliveryType")
    private Integer deliveryType;

    @JsonProperty("externalId")
    private ExternalId externalId;

    @JsonProperty("services")
    private List<ServicesItem> services;

    @JsonProperty("warehouse")
    private Warehouse warehouse;

    @JsonProperty("korobyte")
    private Korobyte korobyte;

    @JsonProperty("places")
    private List<Object> places;

    @JsonProperty("pickupPointCode")
    private String pickupPointCode;

    @JsonProperty("sender")
    private Sender sender;

    @JsonProperty("locationTo")
    private LocationTo locationTo;

    @JsonProperty("maxAbsentItemsPricePercent")
    private Integer maxAbsentItemsPricePercent;

    @JsonProperty("recipient")
    private Recipient recipient;

    @JsonProperty("paymentMethod")
    private Integer paymentMethod;

    @JsonProperty("items")
    private List<ItemsItem> items;

    @JsonProperty("locationFrom")
    private LocationFrom locationFrom;

    @JsonProperty("deliveryCost")
    private Integer deliveryCost;
}
