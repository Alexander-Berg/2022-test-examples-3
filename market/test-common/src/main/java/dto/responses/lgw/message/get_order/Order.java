package dto.responses.lgw.message.get_order;

import java.util.List;
import java.util.Map;

import lombok.Data;

@Data
public class Order {
    private OrderInfo order;

    @Data
    public static class Item {
        private List<Map<String, String>> instances;
    }

    @Data
    public static class OrderInfo {
        private List<Item> items;
    }
}
