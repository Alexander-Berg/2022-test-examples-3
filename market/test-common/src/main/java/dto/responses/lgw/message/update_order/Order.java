package dto.responses.lgw.message.update_order;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class Order {

    @JsonProperty("cargoType")
    private int cargoType;

    @JsonProperty("orderId")
    private OrderId orderId;

    @JsonProperty("amountPrepaid")
    private int amountPrepaid;

    @JsonProperty("deliveryInterval")
    private String deliveryInterval;

    @JsonProperty("shipmentPointCode")
    private String shipmentPointCode;

    @JsonProperty("warehouseFrom")
    private WarehouseFrom warehouseFrom;

    @JsonProperty("total")
    private int total;

    @JsonProperty("tariff")
    private String tariff;

    @JsonProperty("shipmentDate")
    private String shipmentDate;

    @JsonProperty("deliveryDate")
    private String deliveryDate;

    @JsonProperty("cargoCost")
    private int cargoCost;

    @JsonProperty("assessedCost")
    private int assessedCost;

    @JsonProperty("deliveryType")
    private int deliveryType;

    @JsonProperty("services")
    private List<ServicesItem> services;

    @JsonProperty("warehouse")
    private Warehouse warehouse;

    @JsonProperty("korobyte")
    private Korobyte korobyte;

    @JsonProperty("places")
    private List<PlacesItem> places;

    @JsonProperty("sender")
    private Sender sender;

    @JsonProperty("locationTo")
    private LocationTo locationTo;

    @JsonProperty("recipient")
    private Recipient recipient;

    @JsonProperty("paymentMethod")
    private int paymentMethod;

    @JsonProperty("comment")
    private String comment;

    @JsonProperty("items")
    private List<ItemsItem> items;

    @JsonProperty("locationFrom")
    private LocationFrom locationFrom;

    @JsonProperty("deliveryCost")
    private int deliveryCost;
}
