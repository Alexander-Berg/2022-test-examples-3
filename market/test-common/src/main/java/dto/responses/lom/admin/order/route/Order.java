package dto.responses.lom.admin.order.route;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class Order {

    @JsonProperty("openNewTab")
    private Boolean openNewTab;

    @JsonProperty("displayName")
    private String displayName;

    @JsonProperty("id")
    private String id;

    @JsonProperty("slug")
    private String slug;
}
