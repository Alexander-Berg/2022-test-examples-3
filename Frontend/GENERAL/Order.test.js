import React from 'react';
import { shallow } from 'enzyme';

import { Order } from './Order';

describe('Render', () => {
    it('Should render order with default data', () => {
        const wrapper = shallow(
            <Order
                lists={{}}
                onReorder={jest.fn()}
                order={[]}
                missingOrders={[]}
                createOnDeletePerson={jest.fn()}
                createOnReturnPerson={jest.fn()}
                viewValue={'on-duty'}
                onViewChange={jest.fn()}
                viewItems={[{ val: 'on-duty' }, { val: 'not-on-duty' }]}
                ordersLength={0}
                activeOrders={[]}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('Should render order on on-duty mode', () => {
        const wrapper = shallow(
            <Order
                lists={{
                    ['0']: {
                        max: 2,
                        min: 2,
                        items: [
                            {
                                id: 2,
                                login: 'user2',
                                name: { ru: 'Имя2 Фамилия2', en: 'Name2 Surname2' },
                            },
                            {
                                id: 3,
                                login: 'user3',
                                name: { ru: 'Имя3 Фамилия3', en: 'Name3 Surname3' },
                            }],
                    },
                    ['1']: {
                        max: 2,
                        min: 2,
                        items: [
                            {
                                id: 4,
                                login: 'user4',
                                name: { ru: 'Имя4 Фамилия4', en: 'Name4 Surname4' },
                            },
                        ],
                    },
                }}
                onReorder={jest.fn()}
                order={['0', '1']}
                missingOrders={[]}
                createOnDeletePerson={jest.fn()}
                createOnReturnPerson={jest.fn()}
                viewValue={'on-duty'}
                onViewChange={jest.fn()}
                viewItems={[{ val: 'on-duty' }, { val: 'not-on-duty' }]}
                ordersLength={3}
                activeOrders={[
                    {
                        id: 0,
                        login: 'user0',
                        name: { ru: 'Имя0 Фамилия0', en: 'Name0 Surname0' },
                    },
                ]}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    it('Should render order on on-not-duty mode', () => {
        const wrapper = shallow(
            <Order
                lists={{
                    ['0']: {
                        max: 2,
                        min: 2,
                        items: [
                            {
                                id: 2,
                                login: 'user2',
                                name: { ru: 'Имя2 Фамилия2', en: 'Name2 Surname2' },
                            },
                            {
                                id: 3,
                                login: 'user3',
                                name: { ru: 'Имя3 Фамилия3', en: 'Name3 Surname3' },
                            }],
                    },
                    ['1']: {
                        max: 2,
                        min: 2,
                        items: [
                            {
                                id: 4,
                                login: 'user4',
                                name: { ru: 'Имя4 Фамилия4', en: 'Name4 Surname4' },
                            },
                        ],
                    },
                }}
                onReorder={jest.fn()}
                order={['0', '1']}
                missingOrders={[
                    {
                        id: 5,
                        login: 'user5',
                        name: { ru: 'Имя5 Фамилия5', en: 'Name5 Surname5' },
                    },
                ]}
                createOnDeletePerson={jest.fn()}
                createOnReturnPerson={jest.fn()}
                viewValue={'on-not-duty'}
                onViewChange={jest.fn()}
                viewItems={[{ val: 'on-duty' }, { val: 'not-on-duty' }]}
                ordersLength={3}
                activeOrders={[]}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });
});
