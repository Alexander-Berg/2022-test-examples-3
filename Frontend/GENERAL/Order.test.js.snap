// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Should render order on on-duty mode 1`] = `
<div
  className="DutyOrder"
>
  <radio-button
    mix={
      Object {
        "block": "DutyOrder",
        "elem": "filter",
      }
    }
    name="view"
    onChange={[MockFunction]}
    size="s"
    theme="normal"
    value="on-duty"
    view="classic"
  >
    <radio-button__radio
      key="on-duty"
      value="on-duty"
    >
      i18n:on-duty
      <Number>
        3
      </Number>
    </radio-button__radio>
    <radio-button__radio
      key="not-on-duty"
      value="not-on-duty"
    >
      i18n:not-on-duty
      <Number>
        0
      </Number>
    </radio-button__radio>
  </radio-button>
  <ta-title
    level={1}
    mix={
      Object {
        "block": "DutyOrder",
        "elem": "title",
      }
    }
  >
    i18n:now
  </ta-title>
  <Person
    isDnd={false}
    isHoverable={false}
    isNotDeletable={true}
    key="0"
    name="Имя0 Фамилия0"
  />
  <abc-sortable-lists-context
    lists={
      Object {
        "0": Object {
          "items": Array [
            Object {
              "id": 2,
              "login": "user2",
              "name": Object {
                "en": "Name2 Surname2",
                "ru": "Имя2 Фамилия2",
              },
            },
            Object {
              "id": 3,
              "login": "user3",
              "name": Object {
                "en": "Name3 Surname3",
                "ru": "Имя3 Фамилия3",
              },
            },
          ],
          "max": 2,
          "min": 2,
        },
        "1": Object {
          "items": Array [
            Object {
              "id": 4,
              "login": "user4",
              "name": Object {
                "en": "Name4 Surname4",
                "ru": "Имя4 Фамилия4",
              },
            },
          ],
          "max": 2,
          "min": 2,
        },
      }
    }
    onReorder={[MockFunction]}
    order={
      Array [
        "0",
        "1",
      ]
    }
    overflowable={true}
  >
    <Component />
  </abc-sortable-lists-context>
</div>
`;

exports[`Render Should render order on on-not-duty mode 1`] = `
<div
  className="DutyOrder"
>
  <radio-button
    mix={
      Object {
        "block": "DutyOrder",
        "elem": "filter",
      }
    }
    name="view"
    onChange={[MockFunction]}
    size="s"
    theme="normal"
    value="on-not-duty"
    view="classic"
  >
    <radio-button__radio
      key="on-duty"
      value="on-duty"
    >
      i18n:on-duty
      <Number>
        3
      </Number>
    </radio-button__radio>
    <radio-button__radio
      key="not-on-duty"
      value="not-on-duty"
    >
      i18n:not-on-duty
      <Number>
        1
      </Number>
    </radio-button__radio>
  </radio-button>
  <Person
    isDnd={false}
    isHoverable={true}
    isNotDeletable={false}
    key="person-5"
    name="Имя5 Фамилия5"
  />
</div>
`;

exports[`Render Should render order with default data 1`] = `
<div
  className="DutyOrder"
>
  <radio-button
    mix={
      Object {
        "block": "DutyOrder",
        "elem": "filter",
      }
    }
    name="view"
    onChange={[MockFunction]}
    size="s"
    theme="normal"
    value="on-duty"
    view="classic"
  >
    <radio-button__radio
      key="on-duty"
      value="on-duty"
    >
      i18n:on-duty
      <Number>
        0
      </Number>
    </radio-button__radio>
    <radio-button__radio
      key="not-on-duty"
      value="not-on-duty"
    >
      i18n:not-on-duty
      <Number>
        0
      </Number>
    </radio-button__radio>
  </radio-button>
  <Comment>
    i18n:nobody-on-duty
  </Comment>
</div>
`;
